src/factorizations/truncation.jl

```diff
@@ -0,0 +1,230 @@
+# Strategies
+# ----------
+"""
+    TruncationSpace(V::ElementarySpace, by::Function, rev::Bool)
+
+Truncation strategy to keep the first values for each sector when sorted according to `by` and `rev`,
+such that the resulting vector space is no greater than `V`.
+
+See also [`truncspace`](@ref).
+"""
+struct TruncationSpace{S<:ElementarySpace,F} <: TruncationStrategy
+    space::S
+    by::F
+    rev::Bool
+end
+
+"""
+    truncspace(space::ElementarySpace; by=abs, rev::Bool=true)
+
+Truncation strategy to keep the first values for each sector when sorted according to `by` and `rev`,
+such that the resulting vector space is no greater than `V`.
+"""
+function truncspace(space::ElementarySpace; by=abs, rev::Bool=true)
+    isdual(space) && throw(ArgumentError("resulting vector space is never dual"))
+    return TruncationSpace(space, by, rev)
+end
+
+# truncate!
+# ---------
+_blocklength(d::Integer, ind) = _blocklength(Base.OneTo(d), ind)
+_blocklength(ax, ind) = length(ax[ind])
+function truncate_space(V::ElementarySpace, inds)
+    return spacetype(V)(c => _blocklength(dim(V, c), ind) for (c, ind) in inds)
+end
+
+function truncate_domain!(tdst::AbstractTensorMap, tsrc::AbstractTensorMap, inds)
+    for (c, b) in blocks(tdst)
+        I = get(inds, c, nothing)
+        @assert !isnothing(I)
+        copy!(b, @view(block(tsrc, c)[:, I]))
+    end
+    return tdst
+end
+function truncate_codomain!(tdst::AbstractTensorMap, tsrc::AbstractTensorMap, inds)
+    for (c, b) in blocks(tdst)
+        I = get(inds, c, nothing)
+        @assert !isnothing(I)
+        copy!(b, @view(block(tsrc, c)[I, :]))
+    end
+    return tdst
+end
+function truncate_diagonal!(Ddst::DiagonalTensorMap, Dsrc::DiagonalTensorMap, inds)
+    for (c, b) in blocks(Ddst)
+        I = get(inds, c, nothing)
+        @assert !isnothing(I)
+        copy!(diagview(b), @view(diagview(block(Dsrc, c))[I]))
```

Comment:
I find it a bit hard to parse to where the `@view` effect ranges, and I actually think the following is less character(s):


```suggestion
        copy!(diagview(b), view(diagview(block(Dsrc, c)), I))
```